src/ecs/resources/input/input_handler.rs

```diff
@@ -36,8 +36,8 @@ pub struct InputHandler {
     pressed_mouse_buttons: SmallVec<[MouseButton; 16]>,
     down_mouse_buttons: SmallVec<[MouseButton; 16]>,
     released_mouse_buttons: SmallVec<[MouseButton; 16]>,
-    mouse_position: Option<(i32, i32)>,
-    previous_mouse_position: Option<(i32, i32)>,
+    mouse_position: Option<(f64, f64)>,
```

Comment:
Alright so I took a look at the winit git history and did some of my own research to make sure I understood why they were using f64 values.  Some gaming mice are much more precise than your average monitor, so being able to represent "Sub-pixel" movement can actually be helpful if the game is for example a first person shooter.  So with that in mind I'm actually okay with keeping these changes, because a videogame oftentimes has bigger problems to worry about than two f64 values if those values are actually serving a purpose.